src/controllers/assetInfo.ts

```diff
@@ -392,49 +381,36 @@ export class AssetInfoController {
     }
   }
 
-  async updateUsdPrices() {
-    const { usdPrices } = this.store.getState();
+  async updateUsdPricesByAssetIds(assetIds: string[]) {
     const network = this.getNetwork();
 
-    if (!usdPrices || network === NetworkName.Mainnet) {
-      const resp = await fetch(new URL('/api/tickers', MARKETDATA_URL));
-
-      if (resp.ok) {
-        const tickers = (await resp.json()) as Array<{
-          '24h_close': string;
-          amountAssetID: string;
-          priceAssetID: string;
-        }>;
-
-        // eslint-disable-next-line @typescript-eslint/no-shadow
-        const usdPrices = tickers.reduce<Record<string, string>>(
-          (acc, ticker) => {
-            if (
-              !stablecoinAssetIds.has(ticker.amountAssetID) &&
-              ticker.priceAssetID === MARKETDATA_USD_ASSET_ID
-            ) {
-              acc[ticker.amountAssetID] = ticker['24h_close'];
-            }
-
-            return acc;
-          },
-          {}
-        );
+    if (!assetIds.length || network !== NetworkName.Mainnet) {
```

Comment:
I know that `!assetIds.length` and `assetIds.length === 0` check the same thing, but with the latter the intent is much clearer, don't you agree? :-)